gcloud-java-bigquery/src/main/java/com/google/cloud/bigquery/Job.java

```diff
@@ -144,14 +145,62 @@ public boolean isDone() {
   }
 
   /**
+   * Blocks until this job completes its execution, either failing or succeeding. The job status is
+   * checked every 500 milliseconds. This method returns current job's latest information. If the
+   * job no longer exists, this method returns {@code null}.
+   * <pre> {@code
+   * Job completedJob = job.waitFor();
+   * if (completedJob == null) {
+   *   // job no longer exists
+   * } else if (completedJob.status().error() != null) {
+   *   // job failed, handle error
+   * } else {
+   *   // job completed successfully
+   * }}</pre>
+   *
+   * @throws BigQueryException upon failure
+   * @throws InterruptedException if the current thread gets interrupted while waiting for the job
+   *     to complete
+   */
+  public Job waitFor() throws InterruptedException {
+    return waitFor(500, TimeUnit.MILLISECONDS);
+  }
+
+  /**
+   * Blocks until this job completes its execution, either failing or succeeding. The
+   * {@code checkEvery} and {@code unit} parameters determine how often the job's status is checked.
+   * This method returns current job's latest information. If the job no longer exists, this method
+   * returns {@code null}.
+   * <pre> {@code
+   * Job completedJob = job.waitFor(1, TimeUnit.SECONDS);
+   * if (completedJob == null) {
+   *   // job no longer exists
+   * } else if (completedJob.status().error() != null) {
+   *   // job failed, handle error
+   * } else {
+   *   // job completed successfully
+   * }}</pre>
+   *
+   * @throws BigQueryException upon failure
+   * @throws InterruptedException if the current thread gets interrupted while waiting for the job
+   *     to complete
+   */
+  public Job waitFor(int checkEvery, TimeUnit unit) throws InterruptedException {
```

Comment:
I like he var-args option better as it uses the same method signature and can support more options (I actually do think that one may want to say "wait until done or t=x", but with var-args that could always be added later based on demand). However, if in doubt I would only leave the `waitFor()` and add the second option later once we get more feedback.